lib/web_ui/dev/environment.dart

```diff
@@ -112,6 +118,16 @@ class Environment {
     '.dart_tool',
   ));
 
+    /// Path to the ".dart_tool" directory living under `engine/src/flutter`.
+    ///
+    /// This is a designated area for tool downloads which can be used by
+    /// multiple platforms. For exampe: Flutter repo for e2e tests.
+  io.Directory get engineDartToolDir => io.Directory(pathlib.join(
+    engineSrcDir.path,
+    'flutter',
+    '.dart_tool',
```

Comment:
This seems useful for android and iOS e2e too. We currently have duplicated scripts to build AOT/JIT apps, but it would be a win if we can just reuse the tool logic.